packages/api/lambdas/sf-event-sqs-to-db-records/index.js

```diff
@@ -121,6 +121,72 @@ const writeRecords = async ({
   });
 };
 
+function extractCollection(cumulusMessage) {
+  try {
+    return cumulusMessage.meta.collection.name;
+  } catch {
+    return null;
+  }
+}
+
+function extractGranules(cumulusMessage) {
+  try {
+    return cumulusMessage.payload.granules.map((granule) => granule.granuleId);
+  } catch {
+    return null;
+  }
+}
+
+function extractExecution(executionEvent) {
+  try {
+    return executionEvent.detail.executionArn;
+  } catch {
+    return null;
+  }
+}
+
+function extractStateMachine(executionEvent) {
+  try {
+    return executionEvent.detail.stateMachineArn;
+  } catch {
+    return null;
+  }
+}
+
+function extractLambda(executionEvent) {
+  try {
+    return executionEvent.lambda;
+  } catch {
+    return null;
+  }
+}
+
+async function formatCumulusDLQMessage(message, error) {
+  const executionEvent = parseSQSMessageBody(message);
+  const errorString = error.toString();
+  const execution = extractExecution(executionEvent) || 'none';
+  const stateMachine = extractStateMachine(executionEvent) || 'none';
+  let collection;
+  let granule;
+  try {
+    const cumulusMessage = await getCumulusMessageFromExecutionEvent(executionEvent);
+    collection = extractCollection(cumulusMessage) || 'none';
+    granule = extractGranules(cumulusMessage) || 'none';
+  } catch {
+    collection = 'none';
+    granule = 'none';
+  }
+
```

Comment:
the intention with this is to allow backwards compatibility by not changing fields that are already there, but adding new fields that a user *should* be able to just ignore if they're not interested.